Sources/MongoSwift/ChangeStreamEvent.swift

```diff
@@ -1,12 +1,61 @@
 /// An `UpdateDescription` containing fields that will be present in the change stream document for
 /// operations of type `update`.
 public struct UpdateDescription: Codable {
-    /// A document containing key:value pairs of names of the fields that were changed, and the new
-    /// value for those fields.
+    /**
+     * A document containing key:value pairs of names of the fields that were changed (excluding the fields reported
+     * via `truncatedArrays`), and the new value for those fields.
+     *
+     * Despite array fields reported via `truncatedArrays` being excluded from this field, changes to fields of the
+     * elements of the array values may be reported via this field.
+     *
+     * Example:
+     *   original field:
+     *     "arrayField": ["foo", {"a": "bar"}, 1, 2, 3]
+     *   updated field:
+     *     "arrayField": ["foo", {"a": "bar", "b": 3}]
+     *   a potential corresponding UpdateDescription:
+     *     UpdateDescription(
+     *       updatedFields: {
+     *         "arrayField.1.b": 3
+     *       },
+     *       removedFields: [],
+     *       truncatedArrays: [
+     *         TruncatedArrayDescription(
+     *           field: "arrayField",
+     *           newSize: 2
+     *         )
+     *       ]
+     *     )
+     *
+     * Modifications to array elements are expressed via dot notation.
+     * Example: an `update` which sets the element with index 0 in the array field named arrayField to 7 is reported as
+     *   "updatedFields": {"arrayField.0": 7}
+     *
+     * - SeeAlso: https://docs.mongodb.com/manual/core/document/#document-dot-notation
+     */
     public let updatedFields: BSONDocument
 
     /// An array of field names that were removed from the document.
     public let removedFields: [String]
+
+    public struct TruncatedArrayDescription: Codable {
```

Comment:
Can we add a docstring to this? Interesting that swiftlint doesn't automatically flag it, maybe because it's a nested type? I tried it out in a basic file and it seems to catch it though, and I am using the same version of swiftlint as evg :thinking:

Reply:
huh yeah that is weird, was definitely relying too much on swiftlint here to remind me to document it. good catch